### PR TITLE
#96 add tracking of authenticated user via SmtpSession Context 

### PR DESCRIPTION
**Pull request by BrewingCoder**
#96 add tracking of authenticated user via SmtpSession Context 

+ AuthenticatedUser field for session
+ Populate when authenticated

### CODE REVIEW

File: Src/SmtpServer/SmtpSessionContext.cs [hunk 1]
```diff
@@ -72,6 +72,11 @@ internal void RaiseSessionAuthenticated()
         /// </summary>
         public bool IsAuthenticated { get; internal set; }
 
+        /// <summary>
+        /// Returns a value indicating the context's currently authenticated user.  null if no user is authenticated.
+        /// </summary>
+        string AuthenticatedUser { get; internal set; }
```
Comment (cosullivan) [inline Src/SmtpServer/SmtpSessionContext.cs]: Shouldn't this be public?

**Comment by VladimirAkopyan** [comment]
This is quite usefull, can we pull it in and make it public:?

**Comment by cosullivan** [comment]
Hi,

This is already a way to do this by adding the user to the session properties;
https://github.com/cosullivan/SmtpServer/blob/master/Src/SampleApp/Examples/SessionContextExample.cs#L89

However, I've added a new change which I haven't pushed yet which does promote the User name to an Authorization property on the context itself. I will push this out soon.

Thanks,
Cain.

**Comment by cosullivan** [comment]
This is now available.

Thanks,
Cain.

**Pull request closed without merging**